### PR TITLE
chore: Added description about mock for callee

### LINKED ISSUE #305: Missing explanation about mock method
## Overview
If the caller calls the callee function as a mock, the callee function must be implemented on the caller side.
At that time, the function to be implemented is given `#[dynamic_link(CalleeContract, user_defined_mock = true)]`, and the mock to be implemented is `#[cfg(not(target_arch = "wasm32"))]` is given.

An explanation or [link](https://github.com/Finschia/cosmwasm/blob/dynamic_link/packages/derive/src/lib.rs#L167-L171) to the above is needed [here](https://github.com/Finschia/cosmwasm/blob/dynamic_link/contracts/dynamic-caller-contract/src/contract.rs#LL41C7-L41C33).

### PR DESCRIPTION
**Pull request by kokeshiM0chi**
# Description

Closes #305

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [ ] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/Finschia/cosmwasm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly. (not needed)
- [ ] I have added tests to cover my changes. (not needed)
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

**Review by da1suk8 — APPROVED** [review]

File: contracts/dynamic-caller-contract/src/contract.rs [hunk 1]
```diff
@@ -24,6 +24,8 @@ struct CalleeContract {
     address: Addr,
 }
 
+// When `user_defined_mock = true`, user can and must write mock implement
```
Comment (loloicci) [inline contracts/dynamic-caller-contract/src/contract.rs:27]: How about adding "mock implement _for test_"? Is it verbose?

File: contracts/dynamic-caller-contract/src/contract.rs [hunk 2]
```diff
@@ -38,6 +40,7 @@ trait Callee: Contract {
     fn call_caller_address_of(&self, addr: Addr) -> Addr;
 }
 
+// This is the mock implementation specified by `#[cfg(not(target_arch = "wasm32"))]`.
```
Comment (loloicci) [inline contracts/dynamic-caller-contract/src/contract.rs:43]: ditto.

File: contracts/dynamic-caller-contract/src/contract.rs (hunk 1 above)
Comment (kokeshiM0chi) [reply contracts/dynamic-caller-contract/src/contract.rs]: Thank you. Sure, it's better to specify that it's a test. I fixed it like that.

File: contracts/dynamic-caller-contract/src/contract.rs (hunk 2 above)
Comment (kokeshiM0chi) [reply contracts/dynamic-caller-contract/src/contract.rs]: ditto

File: contracts/dynamic-caller-contract/src/contract.rs [hunk 3]
```diff
@@ -24,8 +24,8 @@ struct CalleeContract {
     address: Addr,
 }
 
-// When `user_defined_mock = true`, user can and must write mock implement
-// of the trait for specified struct with `#[cfg(not(target_arch = "wasm32"))]`.
+// When a trait is specified with `user_defined_mock = true`, user can and must write mock implement for test.
```
Comment (loloicci) [inline contracts/dynamic-caller-contract/src/contract.rs:27]: This is not a spec of traits, but a spec of dynamic_link. This comment looks confusing.

**Review by dudong2 — APPROVED** [review]

File: contracts/dynamic-caller-contract/src/contract.rs (hunk 3 above)
Comment (kokeshiM0chi) [reply contracts/dynamic-caller-contract/src/contract.rs]: I fixed as pointed out.

File: contracts/dynamic-caller-contract/src/contract.rs [hunk 4]
```diff
@@ -24,6 +24,8 @@ struct CalleeContract {
     address: Addr,
 }
 
+// When `user_defined_mock = true`, user can and must write mock implement for test
+// of the trait for specified struct with `#[cfg(not(target_arch = "wasm32"))]`.
```
Comment (loloicci) [inline contracts/dynamic-caller-contract/src/contract.rs:28]: which part "with" modifies is ambiguous. rewrite to 

... mock implement with `#[cfg(not(target_arch = "wasm32"))]` of the trait for ...

Comment (kokeshiM0chi) [reply contracts/dynamic-caller-contract/src/contract.rs]: Thanks!! That is certainly true. Fixed.

File: contracts/dynamic-caller-contract/src/contract.rs [hunk 5]
```diff
@@ -24,8 +24,8 @@ struct CalleeContract {
     address: Addr,
 }
 
-// When `user_defined_mock = true`, user can and must write mock implement for test
-// of the trait for specified struct with `#[cfg(not(target_arch = "wasm32"))]`.
+// When `user_defined_mock = true`, user can and must write mock implement
+// with `#[cfg(not(target_arch = "wasm32"))]` for test of the trait for specified struct .
```
Comment (loloicci) [inline contracts/dynamic-caller-contract/src/contract.rs:28]: Unneeded space before "."

**Review by loloicci — APPROVED** [review]

**Pull request merged**